Implemented kruskal algorithm

Michedev opened the pull request:
I've implemented the kruskal algorithm to find the minimum spanning tree in a weighted connected undirected graph. I've wrote two tests for kruskal algorithm and all seems work and i run also your tests after the implementation to check that all works.
I'm not an expert of documentation with python, so please check carefully my docs.

OmkarPathak commented:
The algorithm looks implemented very well. Just a thought, Kruskal's Algorithm is an algorithm of Graph data structures. So, would it be better to place it under `pygorithm/data_structures/graph.py`. If so make the relative changes and  I would be merging the PR. If you think it's a bad idea, comment here so that I can merge PR right away.

Michedev commented:
you have reason, i've moved the implementation into a new _WeightedGraph_ class into graph.py file. I've changed also a little thing into your _Graph_ class: changed the _graph_ variable from _dict_ to _defaultdict_ so the code in _add_edge_ method it's very simplified. I've do some small changes to fit the pep8 style guide into linked_list.py and modules.py. Tell me if you want else i can remove it!
